### ISSUE TITLE
Multiple Base URLs

### ISSUE THREAD

**Issue by saradha-kannan** [androidx/media#799]
This is related to the question: https://github.com/google/ExoPlayer/issues/771

We are in a scenario with where we'd like to have availability zone redundancy on our publish to CDN. The way we want to do this is to publish to primary from AZ 1 and backup from AZ 2. Both manifests will have two BaseURL tags which point to primary and backup playback on CDN.

The main issue we have faced is a synchronization problem between the two ingests. If primary is playing segment 1000 when it fails, it will get a 404 on 1001 and try the backup BaseURL for segment 1001. The problem is that the backup push might have started a few seconds earlier and only be at segment 999 or 1000. We can overcome this by starting them in sync, but it's not desirable.
 
Another issue is if say the primary encoder machine reboots. The encoder will then restart at segment 0. So, if there is ever a need to fail from the backup BaseURL back to the primary BaseURL, we will be looking for some very large segment numbers, but none will be available since it has started from segment 0 (zero) again and will never catch up to the backup segment number.
 
Have you any experience on this in Exoplayer? Wondering if exo player would request the MPD from the new BaseURL that it switches to. Then it would know the exact segment template and not have to rely on the next BaseURL to be in line with the one it switched away from. Because on web currently, dash.js doesn't send request to the backup manifest at all

**Comment by tonihei** [comment]
I don't think this works according to the DASH spec and guidelines. The MPD location is not meant to be affected by `BaseURL` values and the DASH interop guidelines [point this out very clearly](https://dashif-documents.azurewebsites.net/DASH-IF-IOP/master/DASH-IF-IOP.html#mpd-location-feature): "Any present BaseURL element SHALL NOT affect MPD location resolution."

The only official ways to change the MPD location is via a `Location` element or HTTP redirects (see linked section).

**Comment by saradha-kannan** [comment]
Thank you for the reply @tonihei. 
But, my understanding is that Location tag is only used during manifest refresh. If the primary fails, it will not be available to write a new mpd on Location tag. In your experience, how is this kind of fail over usually implemented?

**Comment by tonihei** [comment]
Unfortunately, we are no experts in how to serve media reliably because we only focus on the player implementation. I think you probably need to ask your question elsewhere to get more informative answers. If you think there is something the player should do (while following the specs provided by DASH), please let us know. I'll close this issue for now given there seems to be nothing to do for us.
